ResponseTimes/LogParser.fs

```diff
@@ -4,55 +4,55 @@ open System
 open System.IO
 open FSharp.Data
 open FSharp.Text.RegexProvider
+open Nessos.Streams
 
-type Logs = CsvProvider<"logs.csv"> 
-type Details = JsonProvider<"details.json", SampleIsList=true>
+type Logs = CsvProvider<"sample.csv"> 
+type Details = JsonProvider<"details.json", SampleIsList = true>
 type ResponseTimeRegex = Regex< @"(?<=Request Finished in )((?<ResponseTime>\d+)(?= milliseconds))", noMethodPrefix = true >
 
-type LogDetails = {
-    ApplicationName : string;
-    LoggerName : string;
-    RequestedURL : Uri;
-    HttpResponseStatusCode : int;
-    HttpVerb : string;
-    Message : string;
-    TransactionId : Option<Guid>;
-    ResponseTime : Option<int>;
-}
+/// Only create the matcher once rather than for every call
+let private matcher = ResponseTimeRegex().Match
+
+type LogDetails =
+    { ApplicationName : string
+      LoggerName : string
+      RequestedURL : Uri
+      HttpResponseStatusCode : int
+      HttpVerb : string
+      Message : string
+      TransactionId : Guid option
+      ResponseTime : int option }
 
 let createDetailsJson () =
-    let logs = Logs.Load("logs.csv")
+    let logs = Logs.Load "logs.csv"
     use outFile = new StreamWriter("details.json")
 
-    outFile.WriteLine(sprintf "%s" "[")
-
-    let details = 
-        logs.Rows 
-        |> Seq.take(100000)
-        |> Seq.map(fun log -> log.Details)
-        |> Seq.iter (fun detail -> outFile.WriteLine(sprintf "%s," detail))
+    outFile.WriteLine "["
 
+    logs.Rows 
+    |> Seq.take 100000
+    |> Seq.map(fun log -> log.Details)
+    |> Seq.iter(sprintf "%s," >> outFile.WriteLine)
```

Comment:
a more longhand way would be: -
```fsharp
let formatAndWriteToDisk text = text |> sprintfn "%s," |> outFile.WriteLine
....
Seq.Iter formatAndWriteToDisk
```
The `>>` operator is "compose" i.e. take output of left hand value and pipe into right hand value. So the above can be simplified to: -
```fsharp
let formatAndWriteToDisk = sprintfn "%s," >> outFile.WriteLine
....
Seq.Iter formatAndWriteToDisk
```
And then placed inline as I've done.